vmdb/app/controllers/miq_proxy_controller.rb

```diff
@@ -1053,7 +1053,8 @@ def set_form_vars
     # Get the hosts that have no miq_proxy yet for the Host pulldown list
     if ["new","create"].include?(request.parameters[:action])
       @edit[:avail_hosts] = Hash.new
-      Host.all.delete_if{|h| !h.miq_proxy.blank? || !h.supports_miqproxy?}.sort{|a,b| a.name.downcase<=>b.name.downcase}.each do |host|
+      avail_hosts = Host.all.reject { |h| h.miq_proxy.present? || !h.supports_miqproxy? }
```

Comment:
Prefer using `.reject` over mutating the AR::Relation with `delete_if` because of Rails 4.  cc @tenderlove